src/main/scala/io/hydrosphere/mist/master/MasterService.scala

```diff
@@ -70,19 +78,29 @@ class MasterService(
     for {
       _             <- checkSettings(endpoint)
       context       <- contexts.getOrDefault(req.context)
+      runMode       <- selectRunMode(context, req.workerId)
```

Comment:
It's not necessary to return `Future` in `selectRunMode` - you can use `=` inside `for comprehension`